packages/react-admin-core/src/SelectionRoute.tsx

```diff
@@ -49,8 +49,8 @@ export class SelectionRoute extends React.Component<IProps> {
                             {(props: RouteComponentProps<IRouteParams>) => {
                                 let selectedId: string | undefined;
                                 let selectionMode: "edit" | "add" | undefined;
-                                if (props.match && props.match.params.id === "add") {
-                                    selectedId = undefined;
+                                if (props.match && (props.match.params.id === "add" || props.match.params.id?.startsWith("add-"))) {
```

Comment:
Why not use `add` as a literal in in the route:

```ts
                `${this.match.url}/add/:id`

``` 

instead of manually parsing props.match.params.id ?

Reply:
because the id is optional for add (most (all existing) cases undefined), I would need /add and /add/:id with the problem that sub-routes won't work for /add as /add/:id would match them